[12.0][ADD] stock_orderpoint_impex_matrix, stock_orderpoint_from_inventory

kevinkhao opened the pull request.

kevinkhao committed:
[ADD] stock_orderpoint_impex_matrix, stock_orderpoint_from_inventory

kevinkhao committed:
[IMP] Interfaces change and logic change: check nonempty cells instead of falsy cells (causing problems when values were 0)

sebastienbeau commented:
TODO
- [ ] fix travis
- [ ] improve excel UI (center cell, add some color)
- [ ] add product code

![image](https://user-images.githubusercontent.com/1164578/93490337-a3b7ae80-f908-11ea-8b3d-be1fd778529a.png)

kevinkhao committed:
[WIP] Add default_code column, update matching to default_code
Update column order
Update tests
Update formatting of cells

kevinkhao committed:
Merge branch '12.0-stock_orderpoint_impex' of github.com:akretion/ak-odoo-incubator into 12.0-stock_orderpoint_impex

kevinkhao committed:
[IMP] Points in previous commit,
Some cleanup on constants

kevinkhao committed:
[IMP] Cleanup, Finish points on formatting

kevinkhao committed:
[IMP] Make travis happy

kevinkhao committed:
[IMP] Added and ran pre-commit, make travis happy

kevinkhao committed:
Make travis happy

kevinkhao committed:
Make travis happy

kevinkhao committed:
Make travis happy

kevinkhao commented:
> TODO
> 
> * [ ]  fix travis
> * [ ]  improve excel UI (center cell, add some color)
> * [ ]  add product code
> 
> ![image](https://user-images.githubusercontent.com/1164578/93490337-a3b7ae80-f908-11ea-8b3d-be1fd778529a.png)

Done. UI is changed according to screenshot.
Products are searched on product code instead of product name.
A few fixes on different modules to make travis happy.
Travis seems bugged; everything is working in local but erroring in travis. Investigating.

kevinkhao committed:
[FIX] Make travis happy
Testing on python 3.6 is fine, but before this commit did not work on 3.5 because instantiating a dict did not guarantee correct order

kevinkhao committed:
[FIX] Make travis happy

kevinkhao commented:
(latest version of visual)

![Screenshot from 2020-09-28 09-58-49](https://user-images.githubusercontent.com/19902174/94405736-47a72280-0171-11eb-8694-903f3fcf2204.png)

kevinkhao committed:
[IMP] Add filename to export generation
[IMP] Change string for qty_multiple: Quantité multiple de
[FIX] Fix issues with column order mismatch on displaying and importing
[FIX] Fix unintentional create in case of no matching orderpoint + no values in block, refactor/clean concerned code
[IMP] Clean demo data
[FIX] Merge quantities in case of several lines with same product_id when exporting inventory (lines are merged)
